npm/vite-plugin-cypress-esm/client/moduleCache.js

```diff
@@ -0,0 +1,145 @@
+const __cypressModuleCache = new Map()
```

Comment:
This map is going to become really huge over time, I think we probably want to clear it - maybe on `Cypress:on('test:suite:start')` or something.

There's a bunch of improvements that can likely be made, these will come over time - the goal is to get something that works as expected in the hands of users asap, and we can benchmark/optimize etc as we go.